pivorakmeetup/pivorak-web-app
# Send email with QR after user confirms attendance

### Description

In the previous PR https://github.com/pivorakmeetup/pivorak-web-app/pull/600 I implemented QR code generation, but I forgot to send email with this QR code :)

This PR contains:
- refactoring final decision to execute `VisitRequest::Confirm` & `VisitRequest::Refuse` services
- sending email with QR code after user clicks YES in the confirmation email

### How to test instructions
- Copy production database locally:
`cap production postgres:replicate`
- `$ rails c`
- `$ sidekiq` to deliver emails
- Find last `approved` visit request
`visit_request = VisitRequest.approved.last`
- You will receive confirmation email as on the screenshot
<img width="1280" alt="captura de pantalla 2018-09-30 a las 11 09 31 a m" src="https://user-images.githubusercontent.com/5591973/46255062-52a31b80-c4a1-11e8-9abb-3efd0d26fc60.png">
- Click yes. Then you should receive email with PDF as on the screenshot. Click on `PDF`
<img width="1280" alt="captura de pantalla 2018-09-30 a las 11 10 22 a m" src="https://user-images.githubusercontent.com/5591973/46255075-7cf4d900-c4a1-11e8-98dd-f53dcc34bc11.png">
- Scan QR code from the PDF. It will contain an URL like this - http://localhost:3000/admin/visit_request/<token>/check_in
- Login as `supervisor`
- Send this URL to your computer while you are logged in as `supervisor`
- You should see the following screen:
<img width="1280" alt="captura de pantalla 2018-09-30 a las 11 17 22 a m" src="https://user-images.githubusercontent.com/5591973/46255173-6ef38800-c4a2-11e8-86f3-39de3963dbaa.png">

### Pre-merge checklist
